TypeCobol/Compiler/CodeElements/Expressions/SyntaxValue.cs

```diff
@@ -257,6 +258,7 @@ public virtual bool ValueNeedsCompilationContext
                     case TokenType.IntegerLiteral: // See: GitHub #2054
                     default:
                         System.Diagnostics.Debug.Assert(false, "Unexpected literal value: " + Token.TokenType);
+                        LoggingSystem.LogMessage(LogLevel.Error, "Unexpected literal value: " + Token.TokenType + Environment.NewLine + new System.Diagnostics.StackTrace().ToString());
```

Comment:
This should work. If you prefer, you can restore the original throw, wrap the single usage of the `ValueNeedsCompilationContext` property in a try-catch block and use `LoggingSystem.LogException`. The stacktrace will be automatically included in the trace.